docs/source/trainer/checkpointing.rst

```diff
@@ -379,11 +379,56 @@ There are a few additional trainer arguments which can be helpful to configure:
     will remove each checkpoint after it has been uploaded. Checkpoints are never deleted from object stores.
 
 
-Loading from Object Store
--------------------------
+Saving Checkpoints to Google Cloud Storage (GCS)
+------------------------------------------------
+
+To save checkpoints to GCS, make sure to create a :class:`.RemoteUploaderDownloader` instance
+and pass it to the `Trainer` loggers argument.
+
+.. testcode::
+    :skipif: not _LIBCLOUD_INSTALLED
+
+    from composer.loggers import RemoteUploaderDownloader
+    from composer.trainer import Trainer
+
+    remote_uploader_downloader = RemoteUploaderDownloader(
+        bucket_uri="libcloud://my-gcs-bucket",
+        backend_kwargs={
+            "provider": "google_storage",  # The Apache Libcloud provider name
+            "container": "my-gcs-bucket",  # The name of the cloud container (i.e. bucket) to use.
+            "provider_kwargs": {  # The Apache Libcloud provider driver initialization arguments
+                'key': 'provider_key',  # The cloud provider key.
```

Comment:
should we also have an example of how to provide the key and secret via environment variables? I believe that would be

```
RemoteUploaderDownloader(bucket_uri="libcloud://my-gcs-bucket",
    backend_kwargs={
        "provider": "google_storage",
        "container": "my-gcs-bucket",
        "key_environ": "KEY_ENV_VAR",
        "secret_environ": "SECRET_ENV_VAR"
    }
```

Reply:
I agree with Daniel's point. Is `key` and `secret` a mandatory argument? I think if it is a mandatory argument then user have to change their open source script to pass those key and secret since keeping it in open source world brings a security risk. Is there a way to fetch those keys during runtime by the  `libcloud` ? Like how it happens with AWS key. You add `aws_access_key_id` and `aws_secret_access_key` either in `~/.aws/credentials` file or add it to an instance via envrion variable and the code import those environ variable during runtime so that user doesn't have to provide key manually inside a code.

Reply:
Sorry I wasn't totally clear, the code I gave above _should work today_ (but I haven't actually tried it). That is what is currently implemented for `LibcloudObjectStore`. The AWS stuff works even more smoothly because we have a specific `S3ObjectStore` that uses boto3, instead of the more generic libcloud.

Reply:
also the user could just fetch the key/secret from env vars themselves, which would be essentially the same for them as passing the name of their env vars to us. e.g. `"key": os.environ["KEY_ENV_VAR"]` vs `"key_environ": "KEY_ENV_VAR"`. But both of those should be supported by our current implementation.

Reply:
added example with env variables